test/IssuanceFactory.js

```diff
@@ -40,239 +41,548 @@ const REFUND_ABI = {
+                await factory.participate['address,uint256'](tokenAddress, THOUSAND_CLN, {from: notOwner});
+                // const ccFactoryBalance = await cc.balanceOf(factory.address);
 
 
-        it('should be able to finalize if endtime reached and over softcap (raise cc and cln)', async () => {
-            clnvalue = await cln.balanceOf(accounts[0])
-            await time.increaseTime(10);
-            await cln.approve(Factory.address, THOUSAND_CLN, {from: accounts[1]})
-            assert( await Factory.participate['address,uint256'](tokenAddress, THOUSAND_CLN, {from: accounts[1]}))
-            cc = await ColuLocalCurrency.at(tokenAddress);
-            assert.notEqual(BigNumber(await cc.balanceOf(accounts[1])).toNumber(), 0);
-            await time.increaseTime(10000000);
-            ccvalue = await cc.balanceOf(accounts[0])
-            await Factory.finalize(tokenAddress,{from: accounts[0]})
-            assert.isAbove((await cc.balanceOf(accounts[0])).toNumber(), ccvalue.toNumber() )
-            assert.isAbove((await cln.balanceOf(accounts[0])).toNumber(), clnvalue.toNumber() )
-        });
+                await time.increaseTime(1000000);
+                await factory.finalize(tokenAddress, {from: owner});
 
+                const ownerBalance = await cc.balanceOf(owner);
+                const ccParticipantBalance = await cc.balanceOf(notOwner);
 
+                // transfering CC to factory (by mistake)
+                await cc.transfer(factory.address, ccParticipantBalance, {from: notOwner});
+                assert((await cc.balanceOf(factory.address)).eq(ccParticipantBalance));
 
+                await factory.transferAnyERC20Token(tokenAddress, ccParticipantBalance, {from: owner});
+                assert(ccParticipantBalance.plus(ownerBalance).eq(await cc.balanceOf(owner)));
+            });
+        });
+        describe('CurrencyFactory disabled methods.', async () => {
+            it('insertCLNtoMarketMaker should revert', async () => {
+                await expectRevert(factory.insertCLNtoMarketMaker['address,uint256'](tokenAddress, THOUSAND_CLN, {from: owner}));
+            });
+
+            it('insertCLNtoMarketMaker should revert', async () => {
+                await expectRevert(factory.insertCLNtoMarketMaker['address'](tokenAddress, {from: owner}));
+            });
+
+            it('extractCLNfromMarketMaker should revert', async () => {
+                await expectRevert(factory.extractCLNfromMarketMaker['address,uint256'](tokenAddress, THOUSAND_CLN, {from: owner}));
+            });
+
+            it('extractCLNfromMarketMaker should revert', async () => {
+                await expectRevert(factory.extractCLNfromMarketMaker({from: owner}));
+            });
+
+            it('openMarket should revert', async () => {
+                await expectRevert(factory.openMarket['address'](tokenAddress, {from: owner}));
+            });
+        });
     });
-
 });
```

Comment:
Great job!
* Add tests for the readonly methods
* Add complex 3 sales in parallel with one of them failing